src/qtcore/qml/qml.js

```diff
@@ -209,23 +209,32 @@ function construct(meta) {
     return item;
 }
 
+const typeDefaultValues = {
+  int: 0,
+  real: 0,
+  double: 0,
+  string: '',
+  bool: false,
+  list: [],
+  url: ''
+};
+
 /**
  * Create property getters and setters for object.
  * @param {Object} obj Object for which gsetters will be set
  * @param {String} propName Property name
  * @param {Object} [options] Options that allow finetuning of the property
  */
-function createSimpleProperty(type, obj, propName, options) {
+function createProperty(type, obj, propName, options = {}) {
     var prop = new QMLProperty(type, obj, propName);
     var getter, setter;
 
-    if (typeof options == 'undefined')
-      options = {};
-    else if (typeof options != 'object')
-      options = { default: options }
-
     obj[propName + "Changed"] = prop.changed;
     obj.$properties[propName] = prop;
+    if (!options.hasOwnProperty('default')) {
```

Comment:
@akreuzkamp This could be further improved (even with getting rid of `typeInitialValues`) using some checks like `constructors[newVal.$class].prototype instanceof QMLBaseObject`, but I'm not sure that the other parts there would work fine with that.

We can furhter improve that later, `typeInitialValues` looks like a viable solution for now to me.